controllers/orano2ims_controller.go

```diff
@@ -0,0 +1,600 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/util/retry"
+	k8sptr "k8s.io/utils/ptr"
+
+	oranv1alpha1 "oran-o2ims/api/v1alpha1"
+	"oran-o2ims/controllers/utils"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ORANO2IMSReconciler reconciles a ORANO2IMS object
+type ORANO2IMSReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
```

Comment:
I guess that these `Log` and `Scheme` fields are intended for internal use only. Can they be `logger` and `scheme` to make them private?

Reply:
This is part of the kubebuilder scaffolding and they're also used in cmd/main.go for setting up the controller. I suppose we could try and change these, although all the examples I've looked at in the ACM org have these public:
- https://github.com/stolostron/multiclusterhub-operator/blob/main/controllers/multiclusterhub_controller.go#L74
- https://github.com/stolostron/backplane-operator/blob/main/controllers/backplaneconfig_controller.go#L71
- https://github.com/stolostron/multicluster-observability-operator/blob/main/operators/multiclusterobservability/controllers/multiclusterobservability/multiclusterobservability_controller.go#L81

I propose to leave it as is and update it if it comes out as a concern later on.